src/compiler.rs

```diff
@@ -78,11 +86,16 @@ impl Compiler {
             optional,
             path,
           } => {
+            let expanded_path = match full(&relative.cooked) {
+              Ok(value) => value,
+              Err(error) => return Err(Error::UnresolvedVariableInImport { variable: error.var_name, path: *path }),
+            };
+
             let import = current
               .path
               .parent()
               .unwrap()
-              .join(Self::expand_tilde(&relative.cooked)?)
+              .join(&*expanded_path)
```

Comment:
I somehow doubt `&*` is the "good" thing to do here, even if rust was suggesting it like that :sweat_smile: 
Hints appreciated :slightly_smiling_face:

Reply:
What's the error you're getting otherwise?

Reply:
With just `expanded_path` I get
```
error[E0277]: the trait bound `Cow<'_, str>: AsRef<std::path::Path>` is not satisfied
  --> src/compiler.rs:98:21
   |
98 |               .join(expanded_path)
   |                ---- ^^^^^^^^^^^^^ the trait `AsRef<std::path::Path>` is not implemented for `Cow<'_, str>`
   |                |
   |                required by a bound introduced by this call
   |
   = help: the following other types implement trait `AsRef<T>`:
             <Cow<'_, T> as AsRef<T>>
             <Cow<'_, std::ffi::OsStr> as AsRef<std::path::Path>>
note: required by a bound in `std::path::Path::join`
```
With `&expanded_path` I get the almost identical
```
error[E0277]: the trait bound `Cow<'_, str>: AsRef<std::path::Path>` is not satisfied
  --> src/compiler.rs:98:22
   |
98 |               .join(&expanded_path)
   |                ----  ^^^^^^^^^^^^^ the trait `AsRef<std::path::Path>` is not implemented for `Cow<'_, str>`
   |                |
   |                required by a bound introduced by this call
   |
   = note: required for `&Cow<'_, str>` to implement `AsRef<std::path::Path>`
note: required by a bound in `std::path::Path::join`
  --> /tmp/rustc-1.74.0-src/library/std/src/path.rs:2550:5
help: consider dereferencing here
   |
98 |               .join(&*expanded_path)
   |                      +

For more information about this error, try `rustc --explain E0277`.
```

With `*expanded_path` I get
```
error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> src/compiler.rs:98:21
   |
98 |               .join(*expanded_path)
   |                ---- ^^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |                |
   |                required by a bound introduced by this call
   |
   = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `std::path::Path::join
```

Reply:
I think https://stackoverflow.com/questions/41273041/what-does-combined-together-do-in-rust cleared this up, and this makes sense now